internal/expr/query.go

```diff
@@ -87,17 +90,38 @@ func (pe *PreparedExpr) Query(args ...any) (ce *QueryExpr, err error) {
 		}
 		var val reflect.Value
 		switch tm := typeMember.(type) {
-		case structField:
+		case *structField:
 			val = v.FieldByIndex(tm.index)
-		case mapKey:
+			qargs = append(qargs, sql.Named("sqlair_"+strconv.Itoa(argCount), val.Interface()))
+			argCount++
+		case *mapKey:
 			val = v.MapIndex(reflect.ValueOf(tm.name))
 			if val.Kind() == reflect.Invalid {
 				return nil, fmt.Errorf(`map %q does not contain key %q`, outerType.Name(), tm.name)
 			}
+			if val.Kind() == reflect.Interface {
+				val = val.Elem()
+			}
+			if val.Kind() == reflect.Slice {
```

Comment:
Could this be generalised to handle both `reflect.Slice` and `reflect.Array`?

Reply:
I was considering that, I'll have a go at changing it and see what happens